src/main/java/org/terasology/launcher/platform/Platform.java

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.launcher.platform;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Sets;
+
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * A simplified representation of a computer platform as `os` and `arch`
+ */
+public final class Platform {
+
+    final private OS os;
+    final private Arch arch;
+
+    //TODO(skaldarnar): I'm not fully settled on how to model the Platform. I thought splitting this up into two enums
+    //  for OS and Architecture would help with more type-safe construction of these values, simplify comparison to
+    //  select the right JRE for a game, etc.
+    //
+    //  On the other hand, the set of supported platforms is so limited, and continuing on a non-supported platform
+    //  does not make much sense. So, maybe it is better to have a rather restrictive and explicit enum in the form of
+    //      WINDOWS_X64
+    //      LINUX_X64
+    //  I'm adding the architecture to this list, as I hope that we'll be able to support old Intel and new M1 Macs at
+    //  some point in the future, adding the following to the list:
+    //      MAC_X64
+    //      MAC_AARCH64
+    //  The biggest drawback of being super-strict here is that development on non-supported platforms becomes
+    //  impossible where it was just "not ideal" before.
```

Comment:
It think this should be as type-safe as possible. It shouldn't be possible to run the launcher on unsupported platforms unintentionally. Having seperate enums for `OS` and `Architecture` allows for greater flexibility in adding new platforms in the future and also makes it easier to significantly differentiate by OS alone where needed. You could also possibly make `Platform` itself an enum for further type safety.